### ISSUE TITLE
Bug: CookieStore shouldn't be shard between Context.

### ISSUE THREAD

**Issue by c0per** [jcs224/oak_sessions#19]
Unlink other stores in `src/stores/`, `CookieStore` only hold one `SessionData`. As in readme, `CookieStore` is defined outside the `App` and used in `app.use()`, so that one instance is shared between different contexts (different requests and responses)!

As a result, I'm seeing error like this:
```typescript
TypeError: Cannot read properties of null (reading '_delete')
    at https://deno.land/x/oak_sessions@v3.5.3/src/Session.ts:83:34
    at async dispatch (https://deno.land/x/oak@v10.6.0/middleware.ts:41:7)
...
```
It looks like it's because another concurrent request arrived, session data (in `CookieStore`) gets cleared.
Or even worse, session A can sometimes get the data from session B when using cookieStore.

**Comment by c0per** [comment]
To fix this issue, I think `CookieStore` should be redesigned.

It could hold multiple session data not just one using a map. Then the `CookieStore` looks exactly like `MemoryStore` that falls back to get data from cookie.

**Comment by c0per** [comment]
Plus, the `Session` class itself seems to have the same problem. It shouldn't kept `ctx` passed in from `app.use()` in `this.context`. As the `Session` is defined outside the `App`, it's shared between multiple concurrent `ctx`.

I think `Session` instance should be created for each request, not only one in the entire app. After putting `Session` in `ctx.state.session`, there's no need to store `ctx` in `Session` anyway.

**Comment by c0per** [comment]
Sorry for the tedious comments above, to summarize:

1. `Store` should have one instance per app, and provide data for all the sessions (`CookieStore` currently don't). Since I only want one connection to my redis / db, There's only 1 `Store` instance.
2. `Session` should have one instance per request(`ctx`), (currently one instance per app). Since lots of concurrent requests are going on and they should have separate session.

**Comment by jcs224** [comment]
Thanks for the feedback. A lot to mull over.

Yeah, we should fix the CookieStore for sure.

In general, there's probably a better solution than bootstrapping the whole session instance once. I haven't been able to get a conflicting result in my testing, but I usually am working with either MemoryStore for testing or Postgres for my projects.

**Comment by c0per** [comment]
I'm writing a PR. I hope I can clear things up a bit.

**Comment by jcs224** [comment]
> I'm writing a PR. I hope I can clear things up a bit.

You are awesome :1st_place_medal:

**Comment by jcs224** [comment]
If you are wanting to fix both the session and the CookieStore, do them in separate PRs. Also, it's likely moving the session from once-per-app to once-per-request is going to be a breaking change. But, if you have ideas on how to fix it without a breaking change I'm all for it!

**Comment by jcs224** [comment]
I have an idea for isolating each session without a breaking end-user code (as per the session app context, not CookieStore specifically). I turned on Discussions for this repo and put down my initial thoughts.

https://github.com/jcs224/oak_sessions/discussions/22

**Comment by jcs224** [comment]
Fixed with #23 and #27
